authoring-new-actions.md

```diff
@@ -0,0 +1,130 @@
+
+1. For a JS based action, don&#39;t check node\_modules into master: This is to avoid users to pick dependency on master branch which could be constantly churning. Hence node\_modules shouldn&#39;t be in master. Users are recommended to pick up only releases/v\* version.
+2. Create a release branch for each major version
+3. When ready for a stable release, add a major version tag (example V1)
+4. Create releases for minor and patch version updates (e.g. v1.2.3)
+5. Introduce a new major version of the branch (releases/v2) and tag (v2) if changes break the existing workflow.
+6. For pre-release/preview versions, ensure to name the tag to indicate the same. ex: v0-BETA. There is a system-defined pre-release tag you can leverage while publishing in the marketplace.
+
+![](RackMultipart20201207-4-o60y8a_html_259604dbe9d18c1d.png)
+
+## Publish the action to Marketplace
+
+### Pre-requisites:
+
+- Legal OSS approval process for the repo is complete. Search for the status of the work item created with your repo name here: https://ossmsft.visua lstudio.com/Reviews/\_workitems/ .
+
+&quot;DO NOT PUBLISH THE ACTION TO THE MARKETPLACE UNTIL LEGAL REVIEW IS COMPLETE&quot;
+
+- Squash merge all commits in Master branch and remove all previous git history from your development phase to ensure a clean state when the action is published
+- Versioning guidelines mentioned above section are adhered to and you have a version tag
+- Have a readme.md in place with documentation
+- Include V-team for review process and sign off by mailing@ [GHActions\_vTeam@microsoft.com](mailto:GHActions_vTeam@microsoft.com)
+- Once we are all set, then we make the repo public
+- Finally publish a release to the Marketplace
+
+### Steps to publish:†
+
+- In the GitHub repo, click on Code -\&gt; Releases -\&gt; Draft a new release
+- Enable the checkbox for &quot; **Publish this Action to the GitHub Marketplace**&quot;
+- Fill in the form with Contact &amp; Category details
+  - Typically for all actions intended to &quot;deploy to Azure&quot;, we mark Primary Category as &quot;Deployment&quot;
+  - Secondary Category can be chosen based on the intent of the Action
+- Give a suitable release title and write up. Tip: Include a sample workflow and License details
+- &quot;Publish release&quot;
+- Navigate to [https://github.com/marketplace?type=actions](https://github.com/marketplace?type=actions) and search for your Action to ensure everything is right in place.
+
+### Include a starter template in the [Azure/actions-workflow-sample](https://github.com/Azure/actions-workflow-samples)s repo
```

Comment:
NIT: You could remove `Azure` from the hypertext?

Reply:
> Should we create a docs folder and dump all our HOW-TOs under that directory maybe?

@kaverma to define the TOC and the guidance to enhance this doc with a how-to's section if thats the agreed model.

Reply:
Yes correct. We have a feature in progress for that